App/Containers/ThreadsList.js

```diff
@@ -56,24 +58,59 @@ class ThreadsList extends React.PureComponent {
 
   componentWillMount () {
     // refresh our messages
-    this.props.refreshMessages(true)
+    this.props.refreshMessages()
   }
 
   componentDidMount () {
     this.props.navigation.setParams({
       profile: this.props.profile,
-      hideCreateButton: this.props.tourScreen === true
+      onTour: this.props.tourScreen === true
     })
   }
 
   componentDidUpdate (prevProps, prevState, ss) {
     if (
-      this.props.profile !== prevProps.profile
+      this.props.profile !== prevProps.profile ||
+      this.props.tourScreen !== prevProps.tourScreen
     ) {
       this.props.navigation.setParams({
-        profile: this.props.profile
+        profile: this.props.profile,
+        onTour: this.props.tourScreen === true
       })
     }
+    if (
+      this.props.threads.length !== prevProps.threads.length
+    ) {
+      this._notificationPrompt()
```

Comment:
There's a really nice way to do things like this in a saga. Detaching it from the UI makes it a lot cleaner. Check out the `watchFirstThreeTodosCreation` saga here to see the idea: https://redux-saga.js.org/docs/advanced/FutureActions.html

Reply:
opened a ticket for this, will handle in a follow-up.